File: python/features/steps/policy_steps.py
"""Step definitions for policy feature tests."""

from __future__ import annotations

from behave import given


@given('a policy file "{filename}" with content')
def step_create_policy_file(context, filename: str) -> None:
    """Create a policy file with the given content."""
    policies_dir = context.project_dir / "policies"
    policies_dir.mkdir(exist_ok=True)
    policy_path = policies_dir / filename
    policy_path.write_text(context.text, encoding="utf-8")


@given("no policies directory exists")
def step_no_policies_directory(context) -> None:
    """Ensure no policies directory exists."""
    policies_dir = context.project_dir / "policies"
    if policies_dir.exists():
        import shutil

        shutil.rmtree(policies_dir)


@given("an empty policies directory exists")
def step_empty_policies_directory(context) -> None:
    """Create an empty policies directory."""
    policies_dir = context.project_dir / "policies"
    policies_dir.mkdir(exist_ok=True)
    for policy_file in policies_dir.glob("*.policy"):
        policy_file.unlink()

File: rust/features/steps/policy_steps.rs
//! Step definitions for policy feature tests.

use std::fs;

use cucumber::{gherkin, given};

use crate::step_definitions::initialization_steps::KanbusWorld;

#[given(expr = r#"a policy file {string} with content"#)]
async fn create_policy_file(world: &mut KanbusWorld, step: &gherkin::Step, filename: String) {
    let working_dir = world
        .working_directory
        .as_ref()
        .expect("working directory not set");
    let project_dir = working_dir.join("project");
    let policies_dir = project_dir.join("policies");
    fs::create_dir_all(&policies_dir).expect("failed to create policies directory");

    let policy_path = policies_dir.join(&filename);
    let content = step.docstring.as_ref().expect("policy content required");
    fs::write(&policy_path, content).expect("failed to write policy file");
}

#[given("no policies directory exists")]
async fn no_policies_directory(world: &mut KanbusWorld) {
    let working_dir = world
        .working_directory
        .as_ref()
        .expect("working directory not set");
    let project_dir = working_dir.join("project");
    let policies_dir = project_dir.join("policies");
    if policies_dir.exists() {
        fs::remove_dir_all(&policies_dir).expect("failed to remove policies directory");
    }
}

#[given("an empty policies directory exists")]
async fn empty_policies_directory(world: &mut KanbusWorld) {
    let working_dir = world
        .working_directory
        .as_ref()
        .expect("working directory not set");
    let project_dir = working_dir.join("project");
    let policies_dir = project_dir.join("policies");
    fs::create_dir_all(&policies_dir).expect("failed to create policies directory");

    if let Ok(entries) = fs::read_dir(&policies_dir) {
        for entry in entries.flatten() {
            let path = entry.path();
            if path.extension().and_then(|ext| ext.to_str()) == Some("policy") {
                fs::remove_file(&path).ok();
            }
        }
    }
}
